### PR TITLE
CP - Dynamic Text Substitution 

### PR DESCRIPTION
**Pull request by Arengo**
## Description

Adds CP functionality and the ``<cp>`` DTS for it.

## How Has This Been Tested?

On a custom setup with this clean PR pull.

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: PokeAlarm/Filters.py [hunk 1]
```diff
@@ -47,7 +47,8 @@ def load_pokemon_section(settings):
         "min_sta": 0, "max_sta": 15,
         "quick_move": None, "charge_move": None, "moveset": None,
         "size": None,
-        "gender": None
+        "gender": None,
+        "min_cp": 0, "max_cp": 4760,
```
Comment (kvangent) [inline PokeAlarm/Filters.py]: Can we move min and max cp up above the ivs?

File: PokeAlarm/Manager.py [hunk 2]
```diff
@@ -467,6 +468,19 @@ def process_pokemon(self, pkmn):
                     continue
                 log.debug("Pokemon 'gender' was not checked because it was missing.")
 
+            # Check the Attack IV of the Pokemon
```
Comment (kvangent) [inline PokeAlarm/Manager.py]: Hmm...

File: filters.json.example [hunk 3]
```diff
@@ -20,19 +20,21 @@
             "min_iv":"0", "max_iv":"100", "min_atk": "0", "max_atk":"15",
             "min_def": "0", "max_def":"15", "min_sta": "0", "max_sta":"15",
             "quick_move": null, "charge_move": null, "moveset": null,
-            "size": null, "gender": null, "ignore_missing": "False"
+            "size": null, "gender": null, "min_cp": "0", "max_cp": "4760",
```
Comment (kvangent) [inline filters.json.example]: Lets move CP up before IVs here too

File: filters.json.example [hunk 4]
```diff
@@ -20,19 +20,21 @@
...
+            "ignore_missing": "False"
         },
         "Bulbasaur":[
             {
                 "min_dist":"0", "max_dist":"inf", "min_iv":"0", "max_iv":"100",
                 "min_atk": "0", "max_atk":"0", "min_def": "0", "max_def":"15", "min_sta": "0", "max_sta":"15",
-                "quick_move": null, "charge_move": null, "moveset": null, "size": null, "ignore_missing": "False"
+                "quick_move": null, "charge_move": null, "moveset": null, "size": null, "min_cp": "0", "max_cp": "4760",
```
Comment (kvangent) [inline filters.json.example]: Move it up here too please

File: filters.json.example [hunk 5]
```diff
@@ -20,19 +20,21 @@
...
+                "ignore_missing": "False"
             },
            {
                 "min_dist":"0", "max_dist":"inf", "min_iv":"0", "max_iv":"100",
                 "min_atk": "0", "max_atk":"15", "min_def": "0", "max_def":"15", "min_sta": "0", "max_sta":"15",
                 "quick_move": ["Vine Whip"], "charge_move": ["Sludge Bomb"], "moveset": ["Vine Whip/Sludge Bomb"],
-                "size": null, "ignore_missing": "False"
+                "size": null, "min_cp": "0", "max_cp": "4760", "ignore_missing": "False"
```
Comment (kvangent) [inline filters.json.example]: And here

File: PokeAlarm/Filters.py (hunk 1 above)
Comment (Arengo) [inline PokeAlarm/Filters.py]: Tried to do it across the whole PR.

File: PokeAlarm/Manager.py (hunk 2 above)
Comment (Arengo) [inline PokeAlarm/Manager.py]: This copy pasta.

**Comment by Arengo** [comment]
@kvangent updated requested changes.

**Comment by kvangent** [comment]
Looks good.

**Pull request merged**